crml/staking/src/lib.rs

```diff
@@ -1322,6 +1322,22 @@ impl<T: Trait> Module<T> {
 			.unwrap_or_default()
 	}
 
+	/// Calculate the next accrued payout for a stash id (a payee) without affecting the storage.
+	pub fn accrued_payout(stash: &T::AccountId) -> BalanceOf<T> {
+		let validator_commission_stake_map = Self::current_elected()
```

Comment:
Can you elaborate on your suggestion here?

Reply:
so everytime this RPC is called it will run this calculation, the value of `validator_commission_stake_map` won't change unless era changes or session to be safe e.g. incase of slash.
Seems could be nicer for the node to cache it

Reply:
It would then mean adding another storage item to the staking module, right? Can we do this optimisation later based on the actual load of this RPC on our nodes?

Reply:
Yes not worth blocking the PR over but could see it getting intensive for full nodes: we'd try this: https://substrate.dev/docs/en/knowledgebase/learn-substrate/off-chain-features#off-chain-indexing not onchain